### PR TITLE
chore(grouping): validating Hydration Error works w/o txn

### PR DESCRIPTION
**Pull request by vartec**
Adding a test to validate that when there is no `transaction` tag everything works as expected.
`"<no-value-for-tag-transaction>"` is used as the value of `{{tag.transaction}}`.

Side note: I was unable to repro no transaction scenario using current version NextJS Sentry SDK

### CODE REVIEW

File: tests/sentry/grouping/test_builtin_fingerprinting.py [hunk 1]
```diff
@@ -609,7 +609,7 @@ def test_built_in_hydration_rules_different_transactions(self):
             data=self.hydration_error_trace, project_id=self.project
         )
         data_transaction_text = self.hydration_error_trace.copy()
-        data_transaction_text["transaction"] = "/text/"
+        data_transaction_text["tags"]["transaction"] = "/text/"
```
Comment (vartec) [inline tests/sentry/grouping/test_builtin_fingerprinting.py:612]: @jangjodi FWIW, I'm not sure why this worked before 🤔

File: tests/sentry/grouping/test_builtin_fingerprinting.py [hunk 2]
```diff
@@ -650,3 +650,32 @@ def test_built_in_hydration_rules_disabled(self):
...
+        )
+        variants = {
+            k: v.as_dict()
+            for k, v in event_transaction_no_tx.get_grouping_variants(
+                force_config=GROUPING_CONFIG
+            ).items()
+        }
+
+        # ignore hash as it's not relevant for this test
+        variants["built-in-fingerprint"].pop("hash", None)
```
Comment (armenzg) [inline tests/sentry/grouping/test_builtin_fingerprinting.py:673]: You can use `ANY` if needed.

File: tests/sentry/grouping/test_builtin_fingerprinting.py [hunk 3]
```diff
@@ -650,3 +650,32 @@ def test_built_in_hydration_rules_disabled(self):
...
         assert event.data.data["fingerprint"] == ["my-route", "{{ default }}"]
         assert event.data.data.get("_fingerprint_info") is None
+
+    @with_feature("organizations:grouping-built-in-fingerprint-rules")
+    def test_built_in_hydration_rules_no_transactions(self):
+        """
+        With the flag enabled, hydration errors with no transactions should work as expected.
+        """
+
+        data_transaction_no_tx = self.hydration_error_trace.copy()
```
Comment (armenzg) [inline tests/sentry/grouping/test_builtin_fingerprinting.py:660]: Should this be a fixture?

**Review by armenzg — APPROVED** [review]

**Review by jangjodi — APPROVED** [review]

File: tests/sentry/grouping/test_builtin_fingerprinting.py (hunk 3 above)
Comment (vartec) [reply tests/sentry/grouping/test_builtin_fingerprinting.py:660]: it could, but it's only used in the django TestCase, not in the pytest tests

**Pull request merged**